submissions/Nik3264/Frogger-Game/app.js

```diff
@@ -0,0 +1,186 @@
+
+const gameProperties = {
+  cellHeight: 83,
+  cellWidth: 100,
+  playerXStart: 200,
+  playerYStart: 383,
+  up: 0,
+  down: 383,
+  left: 0,
+  right: 400,
+  collisionX: 50,
+  collisionY: 40,
+  frontierForEnemiesLeft:-150,
+  frontierForEnemiesRight:500,
+};
+
+class Game {
+    constructor({
+      cellHeight,
+      cellWidth,
+      playerXStart,
+      playerYStart,
+      up,
+      down,
+      left,
+      right,
+      collisionX,
+      collisionY,
+      frontierForEnemiesLeft,
+      frontierForEnemiesRight
+    }) {
+      this._level = 1;
+      this.isStop = false;
+      this.isGameOver = false;
+      this.CELL_HEIGHT = cellHeight;
+      this.CELL_WIDTH = cellWidth;
+      this.PLAYER_X_START = playerXStart;
+      this.PLAYER_Y_START = playerYStart;
+      this.up = up;
+      this.down = down;
+      this.left = left;
+      this.right = right;
+      this.collisionX = collisionX;
+      this.collisionY = collisionY;
+      this.frontierForEnemiesLeft=frontierForEnemiesLeft;
+      this.frontierForEnemiesRight=frontierForEnemiesRight;
+    }
+  
+    setLevel() {
+      this._level = 1;
+    }
+    getLevel() {
+      return this._level;
+    }
+    incLevel() {
+      this._level++;
+    }
+    gameStart() {}
+  }
+  
+class Enemy {
+  constructor({ x, y, speed }) {
+    this.sprite = "images/enemy-bug.png";
+    this.x = x;
+    this.y = y;
+    this.speed = speed;
+  }
+
+  update(dt) {
+    this.x +=
+      dt * (this.speed + 0.4 * this.speed * game.getLevel()) * !game.isStop;
+    if (this.x > game.frontierForEnemiesRight) {
+      this.x = game.frontierForEnemiesLeft;
+    }
+    if (this.isCollision()) {
+      player.start();
```

Comment:
Item 4.iii of the requirements says:
> Classes do not refer to any global variables, like global variable player, which is an instance of Player class (referring to global constants and globals provided by the gaming platform like Resources is OK); Hint: pass instance of a game object (or objects) as an argument to other game objects they need to interact with.

It is relevant to this line of code and other methods where global variable `player` is referred to.

Same problem is with variable `game`. Besides, why do you need to refer to a specific instance of class `Game` within methods of that class? I mean you perfectly fine use `this` to refer to the properties of class `Game` and suddenly start using specific instance of the class.

Reply:
I added the player object as a parameter